src/components/Modal.js

```diff
@@ -8,7 +8,10 @@ import SafeHTMLElement from '../helpers/safeHTMLElement';
 export const portalClassName = 'ReactModalPortal';
 export const bodyOpenClassName = 'ReactModal__Body--open';
 
-const canUseDOM = typeof window !== undefined;
+const canUseDOM = typeof window !== 'undefined' || (
+  document && document.createElement !== undefined
```

Comment:
You have checked for `window` existence but you haven't checked for `document` existence therefore if `window` exists and `document` does not then it will throw.
I'd rewrite that as something like:

```js
const canUseDOM = typeof document !== 'undefined' && typeof document.createElement === 'function'
```